store/tikv/2pc.go

```diff
@@ -427,12 +429,22 @@ func (c *twoPhaseCommitter) doActionOnBatches(bo *Backoffer, action twoPhaseComm
 				// Here we makes a new clone of the original backoffer for this goroutine
 				// exclusively to avoid the data race when using the same backoffer
 				// in concurrent goroutines.
-				singleBatchBackoffer := backoffer.Clone()
-				ch <- singleBatchActionFunc(singleBatchBackoffer, batch)
+				singleBatchBackoffer = backoffer.Clone()
 			} else {
-				singleBatchBackoffer, singleBatchCancel := backoffer.Fork()
+				var singleBatchCancel context.CancelFunc
+				singleBatchBackoffer, singleBatchCancel = backoffer.Fork()
 				defer singleBatchCancel()
-				ch <- singleBatchActionFunc(singleBatchBackoffer, batch)
+			}
+			beforeSleep := singleBatchBackoffer.totalSleep
+			ch <- singleBatchActionFunc(singleBatchBackoffer, batch)
+			commitDetail := c.getDetail()
+			if commitDetail != nil {
```

Comment:
When `commitDetail `  will be nil?

Reply:
mainly from pessimistic txn lock:

https://github.com/pingcap/tidb/blob/523b93651d15015c25c8102bb54991f3977ee327/store/tikv/txn.go#L402

https://github.com/pingcap/tidb/blob/523b93651d15015c25c8102bb54991f3977ee327/store/tikv/txn.go#L455

for pessimistic txn, "commit detail" maybe not suitable, maybe better cover pessimistic logic in pessimistic observable task :D

Reply:
Please add a comment here.